Mathlib/Data/Set/Image.lean

```diff
@@ -290,6 +290,13 @@ theorem image_congr' {f g : α → β} {s : Set α} (h : ∀ x : α, f x = g x)
   image_congr fun x _ => h x
 #align set.image_congr' Set.image_congr'
 
+/-- Variant of `image_congr`, for one function being the identity. -/
+theorem image_congr'' {α β : Type*} {f : α → β} {g : β → α} {s : Set α}
+    (h : ∀ x : α, x ∈ s → (g ∘ f) x = x) : g ∘ f '' s = s := by
+  have : ∀ x : α, x ∈ s → (g ∘ f) x = id x := fun x hx  ↦ id_def ▸ h x hx
+  rw [image_congr (this)]
+  aesop
```

Comment:
Can this be something like `simpa [h] using image_congr' (f := g ∘ f) (g := id)`

Reply:
Not really: I could show `have : g ∘ f '' s = id '' s := by simpa [h] using image_congr (f := g ∘ f) (g := id) (s := s) h`, but then I have to simplify id '' s. I've pushed a tweak which does something similar.

One annoying detail is that image_id is only available a few lines below, so I've copied the proof.